test/integration/protocol_integration_test.cc

```diff
@@ -2230,12 +2230,7 @@ TEST_P(DownstreamProtocolIntegrationTest, LargeRequestTrailersRejected) {
 // This test uses an Http::HeaderMapImpl instead of an Http::TestHeaderMapImpl to avoid
 // time-consuming byte size verification that will cause this test to timeout.
 TEST_P(DownstreamProtocolIntegrationTest, ManyTrailerHeaders) {
-  if (GetParam().http1_implementation == Http1ParserImpl::BalsaParser) {
-    // TODO(#21245): Re-enable this test for BalsaParser.
-    return;
-  }
-
-  setMaxRequestHeadersKb(96);
+  setMaxRequestHeadersKb(200);
```

Comment:
I'm supportive of this change (and thank you for the explanation in the description!). Maybe alright to document this slight behavior diff somewhere? I'd be surprised if it caused any practical difference in the wild, but having some list of "known difference" quirks could be helpful just in case.

Reply:
Great idea!  I created a doc and shared it privately.